go/vt/mysqlctl/backup.go

```diff
@@ -334,6 +336,21 @@ func Restore(ctx context.Context, params RestoreParams) (*BackupManifest, error)
 		return nil, err
 	}
 
+	// We disable super_read_only, in case it is in the default MySQL startup
+	// parameters and will be blocking the writes we need to do in
+	// PopulateMetadataTables().  We do it blindly, since
+	// this will fail on MariaDB, which doesn't have super_read_only
+	// This is safe, since we're restarting MySQL after the restore anyway
+	params.Logger.Infof("Restore: disabling super_read_only")
+	if err := params.Mysqld.SetSuperReadOnly(false); err != nil {
+		if strings.Contains(err.Error(), Error1193) {
```

Comment:
@systay I think we have a bigger problem here..
Ideally we would like people to be able to do something like this
```
	errCode := vterrors.Code(err)
	sqlErr, ok := err.(*mysql.SQLError)
```
However, the funcs in the `mysqlctl` package were never migrated to `vterrors`, which means you can't use any of the good stuff you get from that package.
I won't hold up this PR, but something for us to tackle as a cleanup/tech debt task.

Reply:
Created #9311